### ISSUE TITLE
Support extend in SDL schema

### ISSUE THREAD

**Issue by binaryseed** [absinthe-graphql/absinthe#772]
When defining a schema with SDL, the `extend` keyword should add the fields to the existing structure

Example from the spec:
https://graphql.github.io/graphql-spec/draft/#sec-Object-Extensions

```graphql
extend type Query {
  newField: String
}
```

**Comment by mayel** [comment]
What is missing to get extend functionality?

**Comment by benwilson512** [comment]
@mayel basically everything. The parser works I think, but we need a first class entity to model the extension values, and then compilation phases to integrate them in.

**Comment by mayel** [comment]
Thanks. And this PR doesn't do any of that?  https://github.com/absinthe-graphql/absinthe/pull/821

**Comment by binaryseed** [comment]
No, my PR was more of a shortcut, but that turned out not to be sufficient since it didn't treat the extension as a first class concept with structs / etc. It'll probably still be helpful but it needs a different approach

**Comment by mmmries** [comment]
@binaryseed I was just looking into this as part of the work I'm doing on [absinthe_compose](https://hex.pm/packages/absinthe_compose). I'm hoping to support a feature where you can `import_sdl` for an existing server and have proxy all requests for those fields to a single backend, and then sprinkle in other definitions. It would help to support the use case of breaking up a big legacy API by pulling off the different fields one-by-one.

Is there anything I can do to help look into supporting `extend`?

**Comment by binaryseed** [comment]
It's a big task, it will involve adding the `Language` and `Blueprint` extension structs, then using a phase to merge them in when the final schema is being generated.
